### PR TITLE
Add captcha to registration form

### PR DESCRIPTION
**Pull request by gberaudo**
- add the public key associated with the recaptcha dev secret key;
- load google javascript;
- add recaptcha directive and service;
- update auth.html to use captcha for registration.
It is possible to disable captcha for development; see corresponding
setting in API. https://github.com/c2corg/v6_api/pull/268

Related to https://github.com/c2corg/v6_api/issues/218

### CODE REVIEW

**Comment by asaunier** [comment]
Is there something to do to make the CAPTCHA work in a dev instance?
In mine a have the following error:
> ERROR for site owner:
Invalid domain for site key

**Comment by gberaudo** [comment]
What domain name do you use?

**Comment by asaunier** [comment]
c2corgv6-demo.gis.internal

**Comment by gberaudo** [comment]
I see, can you retry now?

**Comment by asaunier** [comment]
It works, thanks!

(the activation link I received after registering seems to be incorrect "invalid nonce" - another problem I guess)

File: c2corg_ui/templates/base.html [hunk 1]
```diff
@@ -121,5 +126,7 @@
          app.sidemenu();
       })();
     </script>
+
+    <script src="https://www.google.com/recaptcha/api.js?onload=vcRecaptchaApiLoaded&render=explicit" async defer></script>
```
Comment (asaunier) [inline c2corg_ui/templates/base.html]: Can't this script be loaded only on the auth page?

Comment (gberaudo) [inline c2corg_ui/templates/base.html]: Yes, done.

**Comment by gberaudo** [comment]
For reference, when testing, the UI url for sending emails must be defined manually in the API configuration.

**Comment by gberaudo** [comment]
Added some documentation about recaptcha in the README.md.

**Comment by asaunier** [comment]
Thanks, looks good to me.

**Comment by gberaudo** [comment]
Thanks for the review.

**Pull request merged**